src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -122,18 +123,28 @@ internal abstract partial class AbstractInitializeMemberFromParameterCodeRefacto
             }
         }
 
-        private IFieldSymbol CreateField(
-            IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts)
+        private async Task<IFieldSymbol> CreateFieldAsync(
+            Document document, IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts, CancellationToken cancellationToken)
         {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var requireAccessibilityModifiers = options.GetOption(CodeStyleOptions.RequireAccessibilityModifiers);
+
             foreach (var rule in rules)
             {
                 if (rule.SymbolSpecification.AppliesTo(SymbolKind.Field, Accessibility.Private))
                 {
                     var uniqueName = GenerateUniqueName(parameter, parameterNameParts, rule);
 
+                    // We do not add an accessibility level if the user turned the setting off
+                    var accessibilityLevel = Accessibility.Private;
+                    if (requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.Never || requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.OmitIfDefault)
+                    {
+                        accessibilityLevel = Accessibility.NotApplicable;
+                    }
```

Comment:
This is not correct for fields in a VB struct.  We want to generate private fields.  And if you do not add the accessibility modifier then the field will be public in that struct.

Reply:
It's actually not just fields.  properties are public in VB as well afaict. Gotta look through all the rules :)

Reply:
You can look at: C:\GitHub\roslyn\src\Features\VisualBasic\Portable\AddAccessibilityModifiers\VisualBasicAddAccessibilityModifiersDiagnosticAnalyzer.vb

from tehre, you can see that teh only time in VB where you get something private is:

```vb
            If member.IsParentKind(SyntaxKind.ClassBlock) OrElse 
               member.IsParentKind(SyntaxKind.ModuleBlock) Then                
                If member.IsKind(SyntaxKind.FieldDeclaration) Then
                    Return accessibility = Accessibility.Private
                End If
            End If
```

So, if the above doesn't hold, you need the accessibility modifier.  If it does hold, hten you can omit if teh user doesn't want redundant modifiers.